metrics/test_driver/cli/common/process/process_wrapper.dart

```diff
@@ -0,0 +1,50 @@
+import 'dart:async';
+import 'dart:io';
+
+/// Base class for [Process] wrappers.
+///
+/// Providers the [stderrBroadcast] and [stdinBroadcast] streams to allow multi subscriptions.
+/// We need to wrap the [Process] because it contains external static methods to start/run
+/// the processes, that are not inheriting in Dart.
+abstract class ProcessWrapper implements Process {
```

Comment:
What about adding a dispose() method here to ensure to close controllers and cancel subscriptions?